packages/clay-drop-down/src/DropDown.tsx

```diff
@@ -75,23 +75,29 @@ const ClayDropDown: React.FunctionComponent<IProps> & {
 	trigger,
 	...otherProps
 }: IProps) => {
-	const triggerElementRef = useRef<HTMLButtonElement>(null);
+	const triggerElementRef = useRef<HTMLButtonElement | null>(null);
 	const menuElementRef = useRef<HTMLDivElement>(null);
 
 	return (
 		<ContainerElement
 			{...otherProps}
 			className={classNames('dropdown', className)}
 		>
-			<span ref={triggerElementRef} style={{display: 'inline-block'}}>
-				{React.cloneElement(trigger, {
-					className: classNames(
-						'dropdown-toggle',
-						trigger.props.className
-					),
-					onClick: () => onActiveChange(!active),
-				})}
-			</span>
+			{React.cloneElement(trigger, {
+				className: classNames(
+					'dropdown-toggle',
+					trigger.props.className
+				),
+				onClick: () => onActiveChange(!active),
+				ref: (node: HTMLButtonElement) => {
+					triggerElementRef.current = node;
+					// Call the original ref, if any
+					const {ref} = trigger as any;
```

Comment:
Is this line necassary, or can we just do this below? Main purpose is to get rid of the `any`

```js
if (typeof trigger.ref === 'function')
```

Reply:
In the end it will give the same error, because there is no `trigger.ref`.

Reply:
hmm I see, so if there is no trigger ref, wont the destructure of it error?

Reply:
I think there will always come an object in the trigger that is the fiber elements and when you don't have a ref will probably declare the ref with undefined.

Reply:
if trigger is undefined, neither will work. But if trigger is an object and ref is undefined, both should work.

this should be the same logically
```js
if (typeof trigger.ref === 'function')
// vs 
const {ref} = trigger;
if (typeof ref === 'function')
```

Unless I am missing something here..

Reply:
Yes both will work you are correct but still we will need `any`. That is what I mean.

so:

```js
if (typeof (trigger as any).ref === 'function')
// ----- 
const {ref} = trigger as any;
if (typeof ref === 'function')
```

Reply:
oh gotcha. didnt know you were referring to the type. yeah for that case I guess its fine. Maybe include an inline comment explaining why we use `any` though. Typically its probably a good practice to add a comment anytime we use `any`

Reply:
Oh yeah, good ideia.